src/streetview/component.js

```diff
@@ -225,6 +234,61 @@ class StreetviewController {
    * Called on initialization of the controller.
    */
   $onInit() {
+    // Initialization
+    if (this.featureStyle) {
+      this.feature_.setStyle(this.featureStyle);
+    }
+
+    /**
+     * @type {string}
+     */
+    const config = this.optionsName ? this.optionsName : 'ngeoStreetviewOptions';
+
+    /**
+     * @type {import('ngeo/options.js').ngeoStreetviewOptions}
+     */
+    this.options = this.injector_.has(config) ? this.injector_.get(config) : null;
+
+    /**
+     * @type {string}
+     */
+    this.viewer = this.options.viewer || 'mapillary';
+
+    if (this.viewer === 'mapillary') {
+      if (!this.options || !this.options.key) {
+        throw new Error('Missing mapillary key');
+      }
+      const clientId = this.options.key;
+      //wait for the mly div to be there before making the service which needs it
+      this.timeout_(() => {
+        const mapillaryService = new MapillaryService(
+          this.scope_,
+          this.map,
+          this.handlePanoramaPositionChange_,
+          clientId
+        );
+        this.scope_.$watch(
+          () => this.panelWidth,
+          (newVal) => {
+            mapillaryService.resize();
+          }
+        );
+        this.streetViewService_ = mapillaryService;
+        this.addWatchers_();
+      });
```

Comment:
Could you please put this whole block of code in a dedicated function `initiMapillary()` (that you call here ?)
Long function are not easy to read but above all, it's more complicated to override by customers if they want to change the code without rewriting all the code for a component.

```suggestion
      this.initMapillary_();
```